awx/main/tests/functional/test_jobs.py

```
@@ -262,3 +309,289 @@ def test_pk_field(self, job_template, organization):
+    with mock.patch('requests.request', return_value=mock_response, autospec=True) as mock_request:
+        task.populate_workload_identity_tokens()
+
+        # Verify the HTTP call was made to the correct endpoint
+        mock_request.assert_called_once()
+        call_kwargs = mock_request.call_args.kwargs
+        assert call_kwargs['method'] == 'POST'
+        assert '/api/gateway/v1/workload_identity_tokens' in call_kwargs['url']
+
+        # Verify context was set on the credential with input source
+        assert 'workload_identity_token' in target_cred.context
+        assert target_cred.context['workload_identity_token'] == 'eyJ.test.jwt'
```

Ruff (0.15.2)

[error] Possible hardcoded password assigned to: "workload_identity_token"

(S105)